packages/docsearch-react/src/DocSearchModal.tsx

```diff
@@ -431,9 +431,11 @@ export function DocSearchModal({
             inputRef={inputRef}
             translations={screenStateTranslations}
             getMissingResultsUrl={getMissingResultsUrl}
-            onItemClick={(item) => {
+            onItemClick={(item, event) => {
               saveRecentSearch(item);
-              onClose();
+              if (!event.shiftKey && !event.ctrlKey && !event.metaKey) {
```

Comment:
We can have a more exhaustive check as we do in our other libraries:

```ts
/**
 * Detect when an event is modified with a special key to let the browser
 * trigger its default behavior.
 */
export function isModifierEvent<TEvent extends MouseEvent | KeyboardEvent>(
  event: TEvent
): boolean {
  const isMiddleClick = (event as MouseEvent).button === 1;

  return (
    isMiddleClick ||
    event.altKey ||
    event.ctrlKey ||
    event.metaKey ||
    event.shiftKey
  );
}
```

Would you mind adding this in `src/utils/isModifierEvent.ts`?

Reply:
Extracted.  :+1:

Reply:
perfect! can you run `yarn lint --fix` please to make the CI pass?

Reply:
Fixed.